File: frontend/src/types/car.ts
export interface CategoryResponse {
  id: number
  name: string
  description?: string
}

export interface CarResponse {
  id: number
  name: string
  brand?: string // Thêm thương hiệu
  price: number
  salePrice?: number // Thêm giá sale (có thể optional)
  stockQuantity: number // Thêm số lượng tồn kho
  description: string
  imageUrl?: string // Thêm link ảnh 2D
  modelUrl: string // VD: /models/ferrari.glb
  color: string
  categoryId: number
  categoryName: string
}

export interface CarRequest {
  name: string
  brand?: string // Thêm thương hiệu
  price: number
  salePrice?: number // Thêm giá sale
  stockQuantity: number // Thêm số lượng tồn kho
  description: string
  imageUrl?: string // Thêm link ảnh 2D
  modelUrl: string
  color: string
  categoryId: number
}

File: backend/demo/src/main/java/com/example/demo/entity/Car.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "cars")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Car {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    // Thương hiệu xe
    private String brand;

    // Giá gốc
    private Double price;

    // Giá khuyến mãi (Có thể null nếu xe không được giảm giá)
    private Double salePrice;

    // Số lượng tồn kho (Bắt buộc phải có để trừ đi khi khách thanh toán)
    @Column(nullable = false, columnDefinition = "integer default 0")
    private Integer stockQuantity;

    @Column(columnDefinition = "TEXT")
    private String description;

    // Ảnh đại diện dạng 2D (Thay thế cho cái icon mặc định trên giao diện)
    private String imageUrl;

    // Đường dẫn file 3D (.glb)
    private String modelUrl;

    // Màu sơn mặc định
    private String color;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;
}